pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -95,6 +101,111 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
-	// Do nothing for now
+	for _, account := range userRecord.Spec.UserAccounts {
+		err = r.ensureUserAccount(reqLogger, account, userRecord)
+		if err != nil {
+			reqLogger.Error(err, "unable to synchronize with member UserAccount")
+			return reconcile.Result{}, err
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileMasterUserRecord) ensureUserAccount(log logr.Logger, recAccount toolchainv1alpha1.UserAccountEmbedded, record *toolchainv1alpha1.MasterUserRecord) error {
+	// get & check fed cluster
+	fedCluster, err := r.getMemberCluster(log, recAccount, record)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(log, record, r.setStatusFailed(targetClusterNotReady), err,
+			"failed to get the member cluster '%s'", recAccount.TargetCluster)
+	}
+
+	// get UserAccount from member
+	nsdName := namespacedName(fedCluster.OperatorNamespace, record.Name)
+	userAccount := &toolchainv1alpha1.UserAccount{}
+	err = fedCluster.Client.Get(context.TODO(), nsdName, userAccount)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// does not exist - should create
+			userAccount = newUserAccount(nsdName, recAccount.Spec)
+			if err := updateStatusConditions(r.client, record, toBeNotReady(provisioningReason, "")); err != nil {
+				return err
+			}
+			if err := fedCluster.Client.Create(context.TODO(), userAccount); err != nil {
+				return err
+			}
+			return nil
+		}
+		return r.wrapErrorWithStatusUpdate(log, record, r.setStatusFailed(unableToGetUserAccountReason), err,
+			"failed to get userAccount '%s' from cluster '%s'", record.Name, recAccount.TargetCluster)
+	}
+
+	// TODO do the synchronization
+	return nil
+}
+
+func (r *ReconcileMasterUserRecord) getMemberCluster(log logr.Logger, recAccount toolchainv1alpha1.UserAccountEmbedded, record *toolchainv1alpha1.MasterUserRecord) (*cluster.FedCluster, error) {
+	// get & check fed cluster
+	fedCluster, ok := r.retrieveMemberCluster(recAccount.TargetCluster)
+	if !ok {
+		return nil, fmt.Errorf("the fedCluster %s not found in the registry", recAccount.TargetCluster)
```

Comment:
Here as it's error message, `Member cluster` goes better than `fed cluster`.

Reply:
changed: 53e9ca6